code/modules/dwarven/dwarven_items.dm

```diff
@@ -0,0 +1,223 @@
+
+/obj/item/dwarven/rune_stone/blitz/apply(atom/target, mob/user)
+	new wall_type(get_turf(user),user)
+	if(user.dir == SOUTH || user.dir == NORTH)
+		new wall_type(get_step(user, EAST),user)
+		new wall_type(get_step(user, WEST),user)
+	else
+		new wall_type(get_step(user, NORTH),user)
+		new wall_type(get_step(user, SOUTH),user)
+	..()
+
+
+/obj/item/dwarven/rune_stone/earth
+	name = "earth runestone"
+	overlay_state = "earth_rune"
+
+/obj/item/dwarven/rune_stone/earth/apply(atom/target, mob/user)
+	var/turf/T = get_turf(target)
+	new /obj/item/pickaxe(T)
+	..()
+
+/obj/item/dwarven/rune_stone/air
+	name = "air runestone"
+	overlay_state = "air_rune"
+
+/obj/item/dwarven/rune_stone/air/apply(atom/target, mob/user)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] fizzles on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		var/teleammount = 0
+		var/teleloc = target
+		if(!isturf(target))
+			teleloc = target.loc
+		for(var/atom/movable/stuff in teleloc)
+			if(!stuff.anchored && stuff.loc && !isobserver(stuff))
+				if(do_teleport(stuff, stuff, 10, channel = TELEPORT_CHANNEL_MAGIC))
+					teleammount++
+					var/datum/effect_system/smoke_spread/smoke = new
+					smoke.set_up(max(round(4 - teleammount),0), stuff.loc) //Smoke drops off if a lot of stuff is moved for the sake of sanity
+					smoke.start()
+	..()
+
+/obj/item/dwarven/mold
+	name = "dwarven mold"
+	desc = "Dwarven mold, one of their great achievments. Allows for casting of very complex tools and armors"
+	icon_state = "mold"
+	w_class = WEIGHT_CLASS_SMALL
+	var/mold_type
+
+/obj/item/dwarven/mallet
+	name = "dwarven mallet"
+	desc = "Dwarven mallet, easy to make , easy to use, other than the fact it is absolutely tiny"
+	icon_state = "mallet"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/dwarven/blueprint
+	name = "dwarven structure print"
+	desc = "Dwarven instructions on how to build a dwarven structure, includes materials how neat."
+	icon_state = "structure_print"
+	w_class = WEIGHT_CLASS_SMALL
+	var/obj/structure/destructible/dwarven/structure
+
+/obj/item/dwarven/blueprint/New(loc,_structure)
+	structure = _structure
+	. = ..()
+
+/obj/item/dwarven/blueprint/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!proximity_flag)
+		return
+	if(isclosedturf(target) ||  isgroundlessturf(target))
+		return FALSE
+	if(!do_after(user, 60, TRUE, user))
+		return FALSE
+	var/turf/place = isopenturf(target) ? target : get_turf(target)
+	new structure(place)
+	qdel(src)
+	. = ..()
+
+/obj/item/dwarven/upgrade_kit
+	name = "dwarven modification kit"
+	desc = "Dwarven instructions on how to modify a weapon."
+	icon_state = "structure_print"
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/dwarven/upgrade_kit/attackby(obj/item/I, mob/living/user, params)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		I.add_creator(H)
+		new /obj/effect/decal/cleanable/ash(get_turf(src))
```

Comment:
```suggestion
		new /obj/effect/decal/cleanable/ash(get_turf(I))
```
To create ash on turf below upgraded item. If we place item on table or turf near us.

Reply:
Uh that would be the same if we did new /obj/effect/decal/cleanable/ash(get_turf(user)). attackedby doesnt check if you attacked it checks if we were attacked